inductiva/resources/machines.py

```diff
@@ -17,73 +19,67 @@ def __init__(
             zone: The zone where the machines will be launched.
         """
         self.id = None
+        self.name = name
         #TODO: Check if machine type is valid.
         self.machine_type = machine_type
         self.num_machines = num_machines
         self.spot = spot
         self.disk_size_gb = disk_size_gb
         self.zone = zone
-        # The negative imply that the cost is unknown.
-        self.estimated_price = -1
 
         # Set the API configuration that carries the information from the client
         # to the backend.
-        self.api_config = inductiva.api.validate_api_key(inductiva.api_key)
+        self._api = instance_api.InstanceApi(api.get_client())
 
     def start(self):
         """Starts a machine group."""
-
-        with inductiva.client.ApiClient(self.api_config) as client:
-            api_instance = inductiva.client.apis.tags.instance_api.InstanceApi(
-                client)
-
-            instance_group_config = \
-                inductiva.client.model.instance_group.InstanceGroup(
-                    machine_type=self.machine_type,
-                    num_instances=self.num_machines,
-                    spot=self.spot,
-                    disk_size_gb=self.disk_size_gb,
-                    zone=self.zone,
-                )
-            try:
-                logging.info("Creating a machine group."
-                             "This may take a few seconds.")
-                start_time = time.time()
-                instance_group = api_instance.create_instance_group(
-                    body=instance_group_config)
-                creation_time_secs = time.time() - start_time
-
-                self.id = instance_group.body["id"]
-                self.name = instance_group.body["name"]
-                #self.estimated_price = self._compute_estimated_price(
-                #    api_instance)
-
-                logging.info("Machine group successfully created in %.2f s.",
-                             creation_time_secs)
-                self._log_machine_group_info()
-
-            except inductiva.client.ApiException as api_exception:
-                raise api_exception
+        instance_group_config = \
+            inductiva.client.model.instance_group.InstanceGroup(
+                machine_type=self.machine_type,
+                num_instances=self.num_machines,
+                spot=self.spot,
+                disk_size_gb=self.disk_size_gb,
+                zone=self.zone,
+            )
+        try:
+            logging.info("Creating a machine group."
+                         "This may take up to a few minutes.")
+            start_time = time.time()
+            instance_group = self._api.create_instance_group(
+                body=instance_group_config)
+            creation_time_mins = (time.time() - start_time) / 60
+
+            self.id = instance_group.body["id"]
+            self.estimated_price = self.estimate_price()
```

Comment:
Is this working already? Can you verify?

Should we still pass it to the user at this moment?